sycl/source/half_type.cpp

```diff
@@ -25,49 +25,39 @@ static uint16_t float2Half(const float &Val) {
   const uint32_t Frac32 = Bits & 0x7fffff;
   // Extract the exponent from the float value
   const uint8_t Exp32 = (Bits & 0x7f800000) >> 23;
-  const int8_t Exp32Diff = Exp32 - 127;
+  const int16_t Exp32Diff = Exp32 - 127;
```

Comment:
Just curious, why the type was changed? if I understand correctly, signed 8-bit integer is capable to store values in range `[-128, 127]`, which should be enough here

Reply:
Exp32 could be 255 (for infinity or NaN) which results in Exp32Diff=128.

Reply:
Ok, thanks for explanation